packages/ember-routing/lib/system/router.js

```diff
@@ -629,6 +633,7 @@ const EmberRouter = EmberObject.extend(Evented, {
 
     let doUpdateURL = function() {
       location.setURL(lastURL);
+      set(emberRouter, 'currentURL', lastURL);
```

Comment:
I think we should actually do this in the `router.updateURL` and `router.replaceURL` methods directly above the `run.once(doUpdateURL)` bit...